modules/http-api/src/main/java/org/mule/runtime/http/api/server/DecodingException.java

```diff
@@ -0,0 +1,21 @@
+/*
+ * Copyright (c) MuleSoft, Inc.  All rights reserved.  http://www.mulesoft.com
+ * The software in this package is published under the terms of the CPAL v1.0
+ * license, a copy of which has been included with this distribution in the
+ * LICENSE.txt file.
+ */
+package org.mule.runtime.http.api.server;
+
+/**
+ * {@code DecodingException} Is an exception thrown when there is attempt to decode a malformed
+ * or invalid text, url or url parameter.
+ *
+ * @since 1.2
+ */
+public class DecodingException extends HttpRequestParsingException {
```

Comment:
Move to the service implementation, everything we put in the API we'll have to maintain backwards compatibility with for years to come. It's best too only move here (the API) necessary bits that require being exposed and keep everything else internal since it allows us to change it as we see fit whenever we want to.